driver-core/src/main/java/com/datastax/driver/core/policies/TokenAwarePolicy.java

```diff
@@ -56,28 +95,36 @@
      * Creates a new {@code TokenAware} policy.
      *
      * @param childPolicy     the load balancing policy to wrap with token awareness.
-     * @param shuffleReplicas whether to shuffle the replicas returned by {@code getRoutingKey}.
-     *                        Note that setting this parameter to {@code true} might decrease the
-     *                        effectiveness of caching (especially at consistency level ONE), since
-     *                        the same row will be retrieved from any replica (instead of only the
-     *                        "primary" replica without shuffling).
-     *                        On the other hand, shuffling will better distribute writes, and can
-     *                        alleviate hotspots caused by "fat" partitions.
+     * @param replicaOrdering the strategy to use to order replicas.
      */
-    public TokenAwarePolicy(LoadBalancingPolicy childPolicy, boolean shuffleReplicas) {
+    public TokenAwarePolicy(LoadBalancingPolicy childPolicy, ReplicaOrdering replicaOrdering) {
         this.childPolicy = childPolicy;
-        this.shuffleReplicas = shuffleReplicas;
+        this.replicaOrdering = replicaOrdering;
     }
 
     /**
-     * Creates a new {@code TokenAware} policy with shuffling of replicas.
+     * Creates a new {@code TokenAware} policy.
      *
-     * @param childPolicy the load balancing policy to wrap with token
-     *                    awareness.
-     * @see #TokenAwarePolicy(LoadBalancingPolicy, boolean)
+     * @param childPolicy     the load balancing policy to wrap with token awareness.
+     * @param shuffleReplicas whether or not to shuffle the replicas.
+     *                        If {@code true}, then the {@link ReplicaOrdering#RANDOM RANDOM} strategy will be used,
+     *                        otherwise the {@link ReplicaOrdering#TOPOLOGICAL TOPOLOGICAL} one will be used.
+     * @deprecated Use {@link #TokenAwarePolicy(LoadBalancingPolicy, ReplicaOrdering)} instead.
+     * This constructor will be removed in the next major release.
+     */
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
+    public TokenAwarePolicy(LoadBalancingPolicy childPolicy, boolean shuffleReplicas) {
+        this(childPolicy, shuffleReplicas ? ReplicaOrdering.RANDOM : ReplicaOrdering.TOPOLOGICAL);
+    }
+
+    /**
+     * Creates a new {@code TokenAware} policy with {@link ReplicaOrdering#TOPOLOGICAL TOPOLOGICAL} replica ordering.
+     *
+     * @param childPolicy the load balancing policy to wrap with token awareness.
      */
     public TokenAwarePolicy(LoadBalancingPolicy childPolicy) {
-        this(childPolicy, true);
+        this(childPolicy, ReplicaOrdering.TOPOLOGICAL);
```

Comment:
For consistency with the existing (3.4.0) behavior, shouldn't this be `ReplicaOrdering.RANDOM`?   Previously we shuffled by default, this would change it so it no longer didn't.